source/world_builder/features/continental_plate_models/grains/random_uniform_distribution_deflected.cc

```diff
@@ -0,0 +1,240 @@
+/*
+  Copyright (C) 2018 - 2021 by the authors of the World Builder code.
+
+  This file is part of the World Builder.
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation, either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "world_builder/features/continental_plate_models/grains/random_uniform_distribution_deflected.h"
+
+#include <algorithm>
+
+#include "world_builder/nan.h"
+#include "world_builder/types/array.h"
+#include "world_builder/types/bool.h"
+#include "world_builder/types/double.h"
+#include "world_builder/types/object.h"
+#include "world_builder/types/one_of.h"
+#include "world_builder/types/unsigned_int.h"
+#include "world_builder/types/value_at_points.h"
+#include "world_builder/utilities.h"
+#include "world_builder/world.h"
+
+#include "world_builder/kd_tree.h"
+
+namespace WorldBuilder
+{
+  using namespace Utilities;
+
+  namespace Features
+  {
+    namespace ContinentalPlateModels
+    {
+      namespace Grains
+      {
+        RandomUniformDistributionDeflected::RandomUniformDistributionDeflected(WorldBuilder::World *world_)
+          :
+          min_depth(NaN::DSNAN),
+          max_depth(NaN::DSNAN)
+
+        {
+          this->world = world_;
+          this->name = "random uniform distribution deflected";
+        }
+
+        RandomUniformDistributionDeflected::~RandomUniformDistributionDeflected()
+          = default;
+
+        void
+        RandomUniformDistributionDeflected::declare_entries(Parameters &prm, const std::string & /*unused*/)
+        {
+          // Document plugin and require entries if needed.
+          // Add compositions the required parameters.
```

Comment:
```suggestion
          // Add compositions to the required parameters.
```

Reply:
Fixed this and some other descriptions of the feature in random_uniform_distribution_deflected